### PR TITLE
Replace nan for null safe to number fns

### PR DESCRIPTION
**Pull request by DavidPhilip**
Still need to check `web-components` and `teams-web-react` - at first glance there shouldn't be any issues

EDIT: on second glance I'm working on a couple of type issues 😅

### CODE REVIEW

**Comment by DavidPhilip** [comment]
@kaisermann Please lmk if this is good to go from your side 🙏🏼

File: src/modules/number.ts [hunk 1]
```diff
@@ -50,9 +52,13 @@ export function formatNumber(
...
-export function clamp(n: number, min: number, max: number): number {
-  if (!Number.isNaN(max) && n > max) return max
-  if (!Number.isNaN(min) && n < min) return min
+export function clamp(
+  n: number,
+  min: number | null,
+  max: number | null,
+): number {
+  if (typeof max === 'number' && n > max) return max
+  if (typeof min === 'number' && n < min) return min
```
Comment (kaisermann) [inline src/modules/number.ts:61]: I'd still keep the `Number.isNaN` checks, just in case 🤷

**Review by kaisermann — APPROVED** [review]

File: src/modules/number.ts (hunk 1 above)
Comment (DavidPhilip) [reply src/modules/number.ts:61]: Good point 👍

Comment (kaisermann) [reply src/modules/number.ts:61]: @DavidPhilip actually, no. Because nothing is bigger or lower than NaN

**Pull request merged**